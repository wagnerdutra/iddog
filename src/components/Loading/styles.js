import styled from 'styled-components';

const ImageStyle = styled.img`
  height: ${props => props.height};
`;

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  background: var(--main-bg-color);
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 999;
  top: 0;
  left: 0;
`;

export { Container, ImageStyle };
