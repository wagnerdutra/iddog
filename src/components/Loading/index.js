import React from 'react';
import gifLoading from 'Root/assets/loader.gif';

import { Container } from './styles';

const Loading = () => <img src={gifLoading} alt="loading" />;

const FullScreenLoading = () => (
  <Container>
    <p>Carregando...</p>
  </Container>
);

export { FullScreenLoading };

export default Loading;
