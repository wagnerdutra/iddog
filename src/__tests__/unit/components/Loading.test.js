import React from 'react';

import { render } from 'react-testing-library';

import Loading, { FullScreenLoading } from 'Components/Loading';

describe('<Loading />', () => {
  it('should render correctly', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});

describe('<FullScreenLoading />', () => {
  it('should render correctly', () => {
    const { container } = render(<FullScreenLoading />);
    expect(container).toMatchSnapshot();
  });
});
