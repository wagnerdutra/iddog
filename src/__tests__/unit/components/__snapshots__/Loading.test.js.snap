// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullScreenLoading /> should render correctly 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  background: var(--main-bg-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 img {
  width: 50px;
  height: 50px;
}

<div>
  <div
    class="c0"
  >
    <p>
      Carregando...
    </p>
  </div>
</div>
`;

exports[`<Loading /> should render correctly 1`] = `
<div>
  <img
    alt="loading"
    class=""
    src="test-file-stub"
  />
</div>
`;
